src/Io/RequestHeaderParser.php

```diff
@@ -103,22 +127,22 @@ private function parseRequest($headers)
 
         // apply REMOTE_ADDR and REMOTE_PORT if source address is known
         // address should always be known, unless this is over Unix domain sockets (UDS)
-        if ($this->remoteSocketUri !== null) {
-            $remoteAddress = \parse_url($this->remoteSocketUri);
+        if ($remoteSocketUri !== null) {
+            $remoteAddress = \parse_url($remoteSocketUri);
             $serverParams['REMOTE_ADDR'] = $remoteAddress['host'];
             $serverParams['REMOTE_PORT'] = $remoteAddress['port'];
         }
 
         // apply SERVER_ADDR and SERVER_PORT if server address is known
         // address should always be known, even for Unix domain sockets (UDS)
         // but skip UDS as it doesn't have a concept of host/port.s
-        if ($this->localSocketUri !== null) {
-            $localAddress = \parse_url($this->localSocketUri);
+        if ($localSocketUri !== null) {
+            $localAddress = \parse_url($localSocketUri);
             if (isset($localAddress['host'], $localAddress['port'])) {
                 $serverParams['SERVER_ADDR'] = $localAddress['host'];
                 $serverParams['SERVER_PORT'] = $localAddress['port'];
             }
-            if (isset($localAddress['scheme']) && $localAddress['scheme'] === 'https') {
+            if (isset($localAddress['scheme']) && $localAddress['scheme'] === 'tls') {
```

Comment:
Yes, this is intentional because we no longer map the transport protocol to the application protocol, see
https://github.com/reactphp/http/pull/346/files#diff-cd234911e405f3c92f4619583db46d3eL166
This is also covered by the test suite 👍

Reply:
👍